# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/fs/parent/datamodel/FileSystemDataManager.h
# Commit: 75dd8ef65751
# Full Hash: 75dd8ef6575143b27c8479af06c86bbe4610e0df
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2022-09-05 21:43:29
# Description:
#   Bug 1789075 - Call RequestAllowToClose on correct thread; r=dom-storage-reviewers,jari
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D156394
# ==============================================================================

diff -r db3dfa11a0ec -r 75dd8ef65751 dom/fs/parent/datamodel/FileSystemDataManager.h
--- a/dom/fs/parent/datamodel/FileSystemDataManager.h	Mon Sep 05 11:58:59 2022 +0000
+++ b/dom/fs/parent/datamodel/FileSystemDataManager.h	Mon Sep 05 12:01:25 2022 +0000
@@ -9,6 +9,7 @@
 
 #include "mozilla/NotNull.h"
 #include "mozilla/TaskQueue.h"
+#include "mozilla/ThreadBound.h"
 #include "mozilla/dom/FileSystemTypes.h"
 #include "mozilla/dom/FileSystemHelpers.h"
 #include "mozilla/dom/quota/CheckedUnsafePtr.h"
@@ -116,7 +117,12 @@
 
   RefPtr<BoolPromise> BeginClose();
 
-  nsTHashSet<FileSystemManagerParent*> mActors;
+  // Things touched on background thread only.
+  struct BackgroundThreadAccessible {
+    nsTHashSet<FileSystemManagerParent*> mActors;
+  };
+  ThreadBound<BackgroundThreadAccessible> mBackgroundThreadAccessible;
+
   const quota::OriginMetadata mOriginMetadata;
   const NotNull<nsCOMPtr<nsISerialEventTarget>> mBackgroundTarget;
   const NotNull<RefPtr<TaskQueue>> mIOTaskQueue;
